content/authoring/guidelines/coding.md

```diff
@@ -7,4 +7,23 @@ next: /authoring/guidelines/sample-tests/
+
+To avoid making kata maintenance a terrible experience, kata code has to follow the guidelines collected below.
+
+:::note
+There are no guidelines which are applicable to every possible case. Some of them have exceptions, or may not be applicable to every kind of task. Some tasks violate them on purpose, like debugging kata or code golf exercises. Sometimes guidelines are difficult to enforce, for example on code which has already been written and poses a big effort to fix. In justified cases, some or all coding guidelines may be omitted, but when it happens, the reason for that should be clearly visible.
+:::
+
+- All code components should follow general [Content Quality Guidelines](/authoring/guidelines/).
+- Once again, **write maintainable code.** Remember that you do not write code for yourself, or for the code runner, but for other users. When writing code, don't play smart and don't be funny.
+- **Every code snippet being a part of a kata should follow the coding guidelines.** The fact that some snippets are not directly visible to the user is not a reason to treat it without appropriate attention.
+- **Stick to rules commonly recognized for your [language](/languages/)**. Know and follow its naming conventions, coding rules, and best practices.
+- **Respect widely recognized standards and conventions related to software design and programming.** Avoid antipatterns and bad practices.
+- **Be familiar with concepts you want to handle and know how to work with them.**
+  - When writing tests, know how unit testing is done, or how to generate random numbers.
+  - Be aware of common pitfalls, like floating point values and comparisons, overflow of arithmetic operations, concatenation of immutable strings, etc.
+  - When your task deals with some specific concepts like concurrency, cryptography, etc. know how they work and how to use them properly.
+- **Write comments** where you think they can be helpful. Put the name of the algorithm used by your reference solution. Explain the structure of tests. Describe your generators of random inputs. This will help maintainers and translators understand and stick to the original idea of your code.
```

Comment:
```suggestion
- **Write comments** where you think they can be helpful. For example, you might include a name, explanation or references for the algorithm used by your reference solution. Explain the structure of tests. Describe your generators of random inputs. This will help maintainers and translators understand and stick to the original idea of your code.
```